### DISCUSSION TITLE
Crash when rendering in an FBO

### DISCUSSION THREAD

**Discussion by acolombier** [pal1000/mesa-dist-win#180] [Q&A] [answered]
Hi,

I am trying to run an open source software called Mixxx, specifically [this branch](https://github.com/acolombier/mixxx/tree/traktor-s4-mk3-dev) on a Windows 11 VM with libvirt. I'm encountering a crash in `d3d11` when calling [QQuickRenderControl::sync](https://doc.qt.io/qt-6/qquickrendercontrol.html#sync). This relies on a FBO to render off-screen sections.

![image](https://github.com/pal1000/mesa-dist-win/assets/7086688/6ae9fe84-7944-45e8-9b66-45123196f5fb)

This is a work in progress feature, so it could very much be that I am misusing the API on my end, but I was wondering if you had some tips on how I could confirm this isn't an issue with mesa.

I am running the program with `MESA_GL_VERSION_OVERRIDE=4.5COMPAT`, on a system-wide mesa install.

I have tried the following:
- Run with `GALLIUM_DRIVER` set to `zink`, `softpipe`, `llvmpipe` and `lavapipe`, but I get no FBO capabilities
- Tried to use other DDL with `QT_OPENGL_DLL` but I either get missing functions or unavailable FBO

Is there some other thing I could try? 

Appreciate your help,
Antoine

**Comment by pal1000** [comment]
`GALLIUM_DRIVER=lavapipe` ls invalid because lavapipe is a Vulkan driver.  That will disable OpenGL as ;long as it's set.

I am not a graphics developer so I recommend asking for help [elsewhere](https://community.khronos.org/).

Also MESA_GL_VERSION_OVERRIDE is no longer necessary with llvmpipe at least, since 23.3.0 as it already advertises OpenGL 4.5 compatibility profile out of the box. For zink and d3d12 I didn't do any tests on feature limitations on current Mesa3D releases.

**Reply by acolombier** [reply, marked as answer]
Thanks for you fast reply! 
It turns out that the issue was with Qt setup, which defaulted to use `d3d11` as a RHI backend, even when the the desktop uses OpenGL. I managed to get output about this after following [Qt troubleshooting instructions](https://doc.qt.io/qt-5/qtquick-visualcanvas-adaptations-d3d12.html#troubleshooting).

Once I set `QSG_RHI_BACKEND=opengl`, I was able to use the Mesa driver, and fixed the crash.

Hopefully this helps someone else! :)
